ghcide/src/Development/IDE/Core/Rules.hs

```diff
@@ -772,9 +777,21 @@ ghcSessionDepsDefinition fullModSummary GhcSessionDepsConfig{..} env file = do
 
             depSessions <- map hscEnv <$> uses_ (GhcSessionDeps_ fullModSummary) deps
             ifaces <- uses_ GetModIface deps
-
-            let            inLoadOrder = map (\HiFileResult{..} -> HomeModInfo hirModIface hirModDetails Nothing) ifaces
-            session' <- liftIO $ mergeEnvs hsc mss inLoadOrder depSessions
+            let inLoadOrder = map (\HiFileResult{..} -> HomeModInfo hirModIface hirModDetails Nothing) ifaces
+#if MIN_VERSION_ghc(9,3,0)
+            mss_imports <- uses_ GetLocatedImports (file : deps)
```

Comment:
A comment explaining why collecting the mod summaries of the direct dependencies here

Reply:
You were right, there was something fishy here. We were including the `ModSummary` for the current module in GHC 9.4 but not before. I've changed this to always include the current `ModSummary`, which allows us to not query imports of direct dependencies. We still need to get the `ModSummary`s of direct dependencies in order to know their `Module`s, which we need to reference in the `ModuleGraph`.

I've written a short note documenting all the assumptions we now make.